### PR TITLE
Add support for Python 3.7 and drop 3.4

### PR DESCRIPTION
**Pull request by hugovk**
Some dependencies need updating to support Python 3.7, including pandas.

There's no release of pandas which supports both 3.4 and 3.7.

So let's drop 3.4, it's very nearly EOL, on 2019-03-16.

https://en.wikipedia.org/wiki/CPython#Version_history

### CODE REVIEW

**Comment by ParthS007** [comment]
@hugovk Can you please resolve conflicts. I have requested a review of @kenneth-reitz and @timofurrer.

Thanks

**Comment by timofurrer** [comment]
That's something Kenneth has to approve.
I'd wait for EOL - 3.4.10. See https://www.python.org/dev/peps/pep-0429/#id4

**Comment by hugovk** [comment]
Python 3.4's EOL is 2019-03-16:

* https://devguide.python.org/#status-of-python-branches

File: docs/intro.rst [hunk 1]
```diff
@@ -76,11 +76,12 @@ Pythons Supported
...
 At this time, the following Python platforms are officially supported:
 
-* cPython 2.7
-* cPython 3.3
-* cPython 3.4
-* cPython 3.5
-* cPython 3.6
+* CPython 2.7
+* CPython 3.3
+* CPython 3.4
```
Comment (claudep) [inline docs/intro.rst]: Why letting 3.3 and 3.4 here?

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tox]
 minversion = 2.4
-envlist = py27, py34, py35, py36
+envlist = py27, py34, py35, py36, py37
```
Comment (claudep) [inline tox.ini]: Remove py34?

File: requirements.txt [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
 requests-toolbelt==0.8.0
-six==1.10.0
-tqdm==4.15.0
+six==1.12.0
+tqdm==4.29.1
 unicodecsv==0.14.1
-urllib3==1.22
-xlrd==1.1.0
+urllib3==1.24.1
+xlrd==1.2.0
```
Comment (claudep) [inline requirements.txt]: I would not have included this upgrade in the same merge request. By the way, I think we should limit this list to direct dependencies. I don't see the advantage of pinning dependencies of dependencies.

File: docs/intro.rst (hunk 1 above)
Comment (hugovk) [reply docs/intro.rst]: Thanks, it was a mistake from conflict resolution. I've removed them now.

File: tox.ini (hunk 2 above)
Comment (hugovk) [reply tox.ini]: Done!

File: requirements.txt (hunk 3 above)
Comment (hugovk) [reply requirements.txt]: It's needed for 3.7, see commit message at https://github.com/kennethreitz/tablib/pull/347/commits/1c3acd87c5b55d92e870560b922dffec925e026f

**Comment by claudep** [comment]
What about rebasing and squashing commits for a cleaner history?

**Comment by hugovk** [comment]
OK, rebased to drop the merge commits.

If you want them as a single commit, you can use the Squash and merge option:

image

**Comment by timofurrer** [comment]
That's fine. Thanks.

We still need BDFL approval in order to merge that. Thanks for your patience!

🎉

**Comment by hugovk** [comment]
The repo has now been transferred to @vinayak-mehta.

@vinayak-mehta Is this okay to merge? It's had three approvals so far. Thanks!

File: .travis.yml [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-  - 3.4
-  - 3.5
-  - 3.6
+matrix:
+  include:
+    - python: 2.7
+    - python: 3.5
+    - python: 3.6
+    - python: 3.7
+      dist: xenial
```
Comment (claudep) [inline .travis.yml]: Travis is currently using xenial by default, I think we can drop this line.

**Comment by claudep** [comment]
Could you please rebase the branch?

**Comment by hugovk** [comment]
Rebased!

Dropped the changes to requirements.txt, they've been dealt with in master.

File: .travis.yml (hunk 4 above)
Comment (hugovk) [reply .travis.yml]: Done!

**Comment by claudep** [comment]
Awesome, thanks!

**Comment by hugovk** [comment]
You're welcome! And thanks for the review and merge!

**Pull request merged**